services/workflows/baseWorkflow.js

```diff
@@ -0,0 +1,172 @@
+"use strict";
+
+const jWorkflow = require("jWorkflow");
+const _ = require("lodash");
+
+module.exports.BaseWorkflow = class BaseWorkflow {
```

Comment:
abstract

Reply:
javascript doesn't support "abstract".
it is the correct way to declare abstract class.